src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -182,15 +195,53 @@
+        } else {
+            final int lastRefInterval2ndBoundary = oneRegion.referenceInterval.getStart();
+            lastReferenceInterval =  new SimpleInterval(oneRegion.referenceInterval.getContig(), lastRefInterval2ndBoundary, refBoundary1stInTheDirectionOfContig);
+        }
+
         final Cigar lastForwardStrandCigar = new Cigar(cigarMemoryList);
+        int clippedNBasesFromEnd = SVVariantCallerUtils.getNumClippedBases(false, oneRegion.cigarAlong5to3DirectionOfContig);
+        if (cigarHasInsertionAsTrailingOperator && cigarElements.get(cigarElements.size()-1).getOperator()==CigarOperator.S) {
+            clippedNBasesFromEnd = cigarElements.get(cigarElements.size()-1).getLength();
+        }
         result.add(new AlignmentRegion(oneRegion.assemblyId, oneRegion.contigId, lastReferenceInterval, lastForwardStrandCigar,
                 oneRegion.forwardStrand, originalMapQ, SVConstants.CallingStepConstants.ARTIFICIAL_MISMATCH,
-                contigIntervalStart, oneRegion.assembledContigLength-SVVariantCallerUtils.getNumClippedBases(false, oneRegion.forwardStrandCigar)));
+                contigIntervalStart, oneRegion.assembledContigLength-clippedNBasesFromEnd));
 
         return result;
     }
 
+    private static boolean parseCigarAndConvertTrailingInsertionToSoftClip(final Cigar cigarAlongInput5to3Direction,
+                                                                           final List<CigarElement> emptyTargetCollection) {
+
+        emptyTargetCollection.addAll(cigarAlongInput5to3Direction.getCigarElements());
+        if (emptyTargetCollection.size()==1) return false;
+
+        if ( (emptyTargetCollection.get(0).getOperator().isClipping() && emptyTargetCollection.get(1).getOperator().isIndel()) ||
+                (emptyTargetCollection.get(emptyTargetCollection.size()-1).getOperator().isClipping() && emptyTargetCollection.get(emptyTargetCollection.size()-2).getOperator().isIndel()) ) {
+            throw new GATKException("Unexpected cigar format with insertion neighboring clipping: " + cigarAlongInput5to3Direction.toString());
+        }
+
+        CigarElement element = emptyTargetCollection.get(0);
+        boolean cigarHasInsertionAsTrailingOperator = false;
+        if (element.getOperator() == CigarOperator.I) {
+            cigarHasInsertionAsTrailingOperator = true;
+            emptyTargetCollection.set(0, new CigarElement(element.getLength(), CigarOperator.S));
+        }
+        element = emptyTargetCollection.get(emptyTargetCollection.size()-1);
+        if (element.getOperator() == CigarOperator.I) {
+            cigarHasInsertionAsTrailingOperator = true;
+            emptyTargetCollection.set(emptyTargetCollection.size()-1, new CigarElement(element.getLength(), CigarOperator.S));
+        }
+
+        return cigarHasInsertionAsTrailingOperator;
```

Comment:
See my comment earlier asking why we can't just return the new cigar list and get rid of this boolean return value.